### PR TITLE
Add an autoscaling FAQ

### PR DESCRIPTION
**Pull request by privatesmith**

### CODE REVIEW

File: docs/faq/autoscaling.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+As the system cannot respond automatically quickly enough in response to sudden
+large traffic spikes, care must be taken when traffic spikes are expected.
+
+There are two common approaches to handling spikes.
+
+The first is to pre-warm the system before performing actions which are likely to
+generate spikes in traffic by manually increasing the number of instances available
+to serve traffic. This approach is useful when the load spike can be predicted
+such as before large customer mailouts or marketing activities.
+
```
Comment (ftsf) [inline docs/faq/autoscaling.md:41]: Link to how to do this here would be good.

File: docs/faq/autoscaling.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+The second is to maintain enough spare capacity in the system to deal with spikes.
+This is the traditional method for dealing with traffic spikes when autoscaling is
+not available and requires less manual intervention, but there is a corresponding
+cost with maintaining the spare capacity.
```
Comment (ftsf) [inline docs/faq/autoscaling.md]: Likewise, instructions for how to do it.

File: docs/faq/autoscaling.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+
+# What is autoscaling?
+
+Autoscaling refers to a system where the resources allocated to an application
+can be dynamically adjusted in response to load on the system.
+
+In Fleet, autoscaling is applied to application server instances (Frontend
+instances). This is designed to allow an application to scale across a
+reasonable range of traffic levels. There will eventually be scalability
```
Comment (ftsf) [inline docs/faq/autoscaling.md]: "eventually" sounds a bit strange. Kind of sounds like "scalability limits" might be a feature we're adding later.

"However the backend database can not autoscale, so an appropriate Fleet size should be chosen to handle your peak load's database requirements." or something.

Comment (privatesmith) [inline docs/faq/autoscaling.md]: That's not how I read it (but I wrote it, so am biased). Will reword.

File: docs/faq/autoscaling.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+limits in regards to the backend (database) for a given size, so the Fleet size
+should be chosen appropriately such that the database can serve the range of
+loads expected.
+
+Autoscaling in Fleet is designed to cope with the normal changes in traffic over a day, but is not
```
Comment (toksvaeth) [inline docs/faq/autoscaling.md:13]: Consider removing 'in Fleet'.  This is a caveat for all autoscaling set ups; not just Fleet's.

File: docs/faq/autoscaling.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+well suited to dealing with sudden spikes in load.
+
+# How does autoscaling work?
+
+In Fleet, autoscaling works by defining high and low watermarks for CPU load
+which define the thresholds at which scaling actions are taken. Scaling actions
+are to either remove a set number of instances at the low watermark, or to
+increase the number of instances by a percentage at the high watermark.
+
+Calculation of the CPU thresholds for purposes of comparison against the
```
Comment (toksvaeth) [inline docs/faq/autoscaling.md:23]: Might be worth mentioning that this is the average across the cluster.

**Pull request merged**